notification/slack/channel.go

```diff
@@ -98,8 +102,18 @@ func (s *ChannelSender) Channel(ctx context.Context, channelID string) (*Channel
 }
 
 func (s *ChannelSender) loadChannel(ctx context.Context, channelID string) (*Channel, error) {
+	var err error
 	cfg := config.FromContext(ctx)
 
+	teamID, ok := s.teamIDCache.Get(cfg.Slack.AccessToken)
+	if !ok {
+		teamID, err = lookupTeamIDForToken(ctx, cfg.Slack.AccessToken)
+		if err != nil {
+			return nil, err
+		}
+		s.teamIDCache.Add(cfg.Slack.AccessToken, teamID)
+	}
```

Comment:
We should wrap this in a mutex so we only fetch once if we get concurrent calls. Also, we could probably drop the `lru` usage since it doesn't get us much.
```suggestion
	s.teamMx.Lock()
	if s.teamID == "" || s.token != cfg.Slack.AccessToken {
		// token changed or teamID missing
		id, err := lookupTeamIDForToken(ctx, cfg.Slack.AccessToken)
		if err != nil {
			s.teamMx.Unlock() // always unlock
			return nil, err
		}
		
		// update teamID and token after fetching succeeds
		s.teamID = id
		s.token = cfg.Slack.AccessToken
	}
	teamID := s.teamID
	// unlock after copying current teamID
	s.teamMx.Unlock()
```